src/reducer/resource/list/ids.js

```diff
@@ -1,10 +1,62 @@
-import { CRUD_GET_LIST_SUCCESS, CRUD_DELETE_SUCCESS } from '../../../actions/dataActions';
+import { CRUD_GET_MANY, CRUD_GET_LIST_SUCCESS, CRUD_DELETE_SUCCESS } from '../../../actions/dataActions';
 
-export default resource => (previousState = [], { type, payload, requestPayload, meta }) => {
+const cacheDuration = 10 * 60 * 1; // ten minutes
+
+/**
+ * Add new records to the pool, and remove outdated ones.
+ *
+ * This is the equivalent of a stale-while-revalidate caching strategy:
+ * The cached data is displayed before fetching, and stale data is removed
+ * only once fresh data is fetched.
+ */
+const addRecords = (newRecords = [], oldRecords) => {
```

Comment:
This looks like a big copy-paste of `data.js`. Besides, why is it related to the clear filter bug?